transaction/core/Cargo.toml

```diff
@@ -52,6 +52,7 @@ features = ["default-code-coverage"]
 curve25519-dalek = { version = "4.0.0-pre.2", default-features = false, features = ["nightly", "u64_backend"] }
 
 [dev-dependencies]
+assert_matches = "1.5"
```

Comment:
Why do you need this? `assert!(matches!(` worked in other places of the codebase and didn't require another dependency.

Reply:
i think `assert_matches` is nicer because `assert!(matches!(`, when it fails, just tells you "matches! was false",
but `assert_matches!` will be able to log the expression that failed to match which helps you debug it